tests/primeqa/services/rest_server/test_server.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright 2022 PrimeQA Team
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import List
+from unittest.mock import MagicMock
+import pytest
+from fastapi import FastAPI
+from fastapi.testclient import TestClient
+# from primeqa.services.rest_server.server import FastAPI
+import sys
+import os
+from primeqa.services.configurations import Settings
+from primeqa.services.rest_server.data_models import Reader
+from primeqa.services.rest_server.server import app
+
+
+class TestServer:
+
+    @pytest.fixture()
+    def client(self):
+        return TestClient(app)
+
+    @pytest.fixture()
+    def mock_STORE(self, mocker) -> MagicMock:
+        return mocker.patch(
+            "primeqa.services.rest_server.server.STORE",
+            autospec=True,
+        )
+
+    def test_get_readers(self, client):
+        response = client.get("/readers")
+        assert response.status_code == 200
+
+    def test_get_answers(self, client):
+        response = client.post(
+            "/answers",
+            json={
+                "question": "Test",
+                "contexts": [["Example text with test word in a context list"]],
+                "queries": ["Example text queries prop"],
+                "reader": {
+                    "reader_id": "ExtractiveReader",
+                    "parameters": []
+                }
+            },
+        )
+        assert response.status_code == 201, response.text
+
+    def test_answer_with_empty_question(self, client):
+        response = client.post(
+            "/answer",
+            json={
+                "question": "",
+                "retriever": {"retriever_id": "test retriever"},
+                "collection": {"collection_id": "test collection"},
+                "reader": {"reader_id": "test reader"},
+            },
+        )
+        assert response.status_code == 404, response.text
+        assert response.json() == {"detail": "Not Found"}
+
+    def test_get_indexers(self, client):
+        response = client.get("/indexers")
```

Comment:
1. Refer above about adding a mocked registry. 
2. Along with Registry I would mock the Indexer Instance also to avoid loading model and making tests heavier since that functionality is covered under different unit test.
3. I would also add `assert` on response from mocked registry to see that JSON is structured properly on response.